### PR TITLE
Enable more psalm rules

### PR DESCRIPTION
**Pull request by provokateurin**
## Summary

Enables a few more useful rules:
- MissingThrowsDocblock: Ensures every method has the appropriate `@throws` annotations
- InvalidOperand: Ensures values/variables of different types are not used in an operation without explicit casting (e.g. `'test' . 123`
- UnusedPsalmSuppress: Reports on unused supresses

I only fixed the problems of the last rule because there weren't a lot, but the first two ones I added to the baseline. Those errors will only show up for new or refactored code.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by nickvergessen — CHANGES_REQUESTED** [review]
> MissingThrowsDocblock: Ensures every method has the appropriate @throws annotations

I do not agree with this.
Only expected throws should be documented.
Any method that does do anything with the DB, Redis, Ldap, ... would otherwise document to throw them and every consumer starts to wrap hundreds of catches or has their IDE crying about unhandled exceptions.

**Review by st3iny — CHANGES_REQUESTED** [review]
I agree with Joas. Explicit error handling like in Go and Rust is nice but it's already to late to enforce that in our codebase.

Only relevant exceptions should be listed to make the caller aware of possible errors. The decision is up to the developer.

File: core/Controller/OCMController.php [hunk 1]
```diff
@@ -58,6 +58,7 @@ public function __construct(
 	 *
 	 * @PublicPage
 	 * @NoCSRFRequired
+	 * @psalm-suppress UnusedPsalmSuppress
```
Comment (come-nc) [inline core/Controller/OCMController.php:61]: :upside_down_face:

Comment (provokateurin) [reply core/Controller/OCMController.php:61]: Yeah this piece of code is somehow very broken :grimacing:

**Pull request closed without merging**